manifests/unit_file.pp

```diff
@@ -46,6 +46,9 @@
 # @param active
 #   If set, will manage the state of the unit.
 #
+# @param restart
+#   Specify a restart command manually. If left unspecified, the service will be stopped and then started.
```

Comment:
I read https://puppet.com/docs/puppet/7.3/type.html#service-attribute-restart and I think it's outdated. if `hasrestart` is true, the restart command is used. Note that since Puppet 2.7 `hasrestart` defaults to true.

Perhaps you can avoid it this way:

```suggestion
#   Specify a restart command manually. If left unspecified, a standard Puppet service restart happens.
```

For reference:

Reply:
the standard Puppet service restart uses command `systemctl restart <service>` that stops and than starts the service.

Reply:
That is the implementation, but at the module level we can't say much more than "we'll do what Puppet does".